src/main/java/seedu/address/logic/parser/ReminderCreateCommandParser.java

```diff
@@ -0,0 +1,65 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATETIME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DESCRIPTION;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.ReminderCreateCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.reminder.DateTime;
+
+
+/**
+ * Parses input arguments and creates a new ReminderCreateCommand object
+ */
+public class ReminderCreateCommandParser implements Parser<ReminderCreateCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the CreateTagCommand
+     * and returns a CreateTagCommand object for execution
+     * @throws ParseException if the user input does not conform to the specified format
+     */
+    public ReminderCreateCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_DESCRIPTION, PREFIX_DATETIME);
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_DESCRIPTION, PREFIX_DATETIME)) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    ReminderCreateCommand.MESSAGE_USAGE));
+        }
+        Optional<Index> index;
+
+        try {
+            index = parseIndex(argMultimap);
+        } catch (ParseException pe) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    ReminderCreateCommand.MESSAGE_USAGE), pe);
+        }
+
+        String description = argMultimap.getValue(PREFIX_DESCRIPTION).get();
+        DateTime dateTime = ParserUtil.parseDateTime(argMultimap.getValue(PREFIX_DATETIME).get());
+        return index.map(i -> new ReminderCreateCommand(i, description, dateTime))
+                .orElse(new ReminderCreateCommand(description, dateTime));
+    }
+
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
+        return Stream.of(prefixes).allMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
+    }
+
+    private Optional<Index> parseIndex(ArgumentMultimap argMultimap) throws ParseException {
```

Comment:
Why not use the parseIndex in parserutil?

Reply:
The parseIndex in ParserUtil is called within this function if the user input contains an index, otherwise, there will be no need to parse the index (target person).